### PR TITLE
Update `truffle init` artifacts

### PR DESCRIPTION
**Pull request by sukanyaparashar**
### ISSUE
Update `truffle init` artifacts. See #5268.

### SOLUTION
This PR follows the below mentioned steps to update `truffle init` artifacts - 

- [x] Remove `Migrations.sol` from `contracts`.
- [x] Remove `1_initial_migration.js` from `migrations`.
- [x] Update the sample `truffle-config.js` including the following -
  - [x] Use Goerli or Sepolia as the testnet.
  - [x] Change `.secret` to `.env`.
  - [x] Get `mnemonic` and `Infura project id` from `.env`.
  - [x] The comment says auto create at `9545`, but the development environment config is `8545`. What should be the ideal behaviour? I guess, the comment needs to be changed and use `managed Ganache instance` instead of `development blockchain`.
- [x] Add a `.gitignore` with `.env` in it.

### CODE REVIEW

File: packages/core/lib/commands/init/initSource/.gitignore
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/.gitignore:1]: Let's have a newline at the end.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:29]: Since the default we give users is reading from `process.env`, that means we want users to use dotenv right? So I would phrase this as:

_It is recommended to store your secret variables in a_ `.env` _file (which you should include in_ `.gitignore`_). In your project root, run_ `npm i dotenv` _, create_ `.env` _and declare your MNEMONIC and infura PROJECT\_ID variables inside._

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:70]: Does the comment about gas hold true for Goerli?

Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:70]: Does the comment about gas hold true for Goerli?

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:92]: This indent should stay.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:102]: This comma should stay.

**Review by cliffoo — CHANGES_REQUESTED** [review]
Thanks @sukanyaparashar for taking this on! I've left a few comments. Note that [issue#5268](https://github.com/trufflesuite/truffle/issues/5268) also mentioned changing the comment on line 35 in `truffle-config.js` (from port `9545` to `8545`)

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [reply packages/core/lib/commands/init/initSource/truffle-config.js:29]: Also destructure maybe? `const { MNEMONIC, PROJECT_ID } = process.env;`

Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:29]: Your suggestions make sense to me. I will change the comments. Thanks @cliffoo.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:92]: Yeah! I forgot to add `--no-verify` tag in the commit to get pass the prettier. I will add that again. Thanks for catching this @cliffoo.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:102]: `prettier` again :(

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cds-amal) [reply packages/core/lib/commands/init/initSource/truffle-config.js:92]: Please don't use the `--no-verify` flag when committing, as that's the process.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cds-amal) [reply packages/core/lib/commands/init/initSource/truffle-config.js:102]: Same, please don't use `--no-verify` flag on commits.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:92]: `--no-verify` flag is only used with `truffle-config.js` when committing due to the fact that `prettier` removes the trailing commas in that file, which is required for user convenience. There should be a way to exclude this file from using `prettier`, which I guess can be a separate PR to resolve this issue.

**Comment by sukanyaparashar** [comment]
> Thanks @sukanyaparashar for taking this on! I've left a few comments. Note that [issue#5268](https://github.com/trufflesuite/truffle/issues/5268) also mentioned changing the comment on line 35 in `truffle-config.js` (from port `9545` to `8545`)

Thanks for the review @cliffoo. In regards to the point of changing the port from `9545` to `8545`, there is a catch here. Port `9545` is used by the managed **Ganache** when `truffle develop` is run with no network config. 

On the other hand, when `truffle test` is run, it looks for the `development` network settings in the config by default, where `8545` is mentioned if `ganache-cli` is used as the network (Note: **ganache-cli** has the default port `8545`).

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (eggplantzzz) [reply packages/core/lib/commands/init/initSource/truffle-config.js:102]: So what should you do when committing non-JS files or in situations like this where you don't want prettier to update things? I think in this case using that flag should be fine. Do you disagree or have a different solution?

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (eggplantzzz) [reply packages/core/lib/commands/init/initSource/truffle-config.js:70]: I'm not sure if it does and maybe this is not very helpful in here. What do you think? Should we just get rid of this `gas` line altogether?

Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:70]: I couldn't find any suitable documentation for this `gas` line. I guess we can just get rid of this for now.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:102]: We were discussing that if we can somehow exclude some files from using prettier. I have raised an issue for this https://github.com/trufflesuite/truffle/issues/5360.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [reply packages/core/lib/commands/init/initSource/truffle-config.js]: @gnidan @haltman-at Do you guys have any insight on where this `gas` comment is from? Should it be kept / changed now that it's goerli?

**Comment by cliffoo** [comment]
@sukanyaparashar Yea I think the `development` network config is fine, the confusion comes from [here](https://github.com/trufflesuite/truffle/blob/1dce4898635c2d4264d0fde2a3ffa3ba5a37ba37/packages/core/lib/commands/init/initSource/truffle-config.js#L34), where it mentions a `"development blockchain"`, which makes one wonder if it's referring to the `development` network (it isn't). I don't really know how to call it instead, `"dev"`? `"ephemeral"`? Your call 😅

**Comment by sukanyaparashar** [comment]
> @sukanyaparashar Yea I think the `development` network config is fine, the confusion comes from [here](https://github.com/trufflesuite/truffle/blob/1dce4898635c2d4264d0fde2a3ffa3ba5a37ba37/packages/core/lib/commands/init/initSource/truffle-config.js#L34), where it mentions a `"development blockchain"`, which makes one wonder if it's referring to the `development` network (it isn't). I don't really know how to call it instead, `"dev"`? `"ephemeral"`? Your call 😅

I guess a better name will be `"managed Ganache instance"` instead of a `"development blockchain"`, may be?

**Comment by cliffoo** [comment]
Sounds fine by me 👍 It's nice that it maps to the variable name in code.

**Review by eggplantzzz — APPROVED** [review]
cooool - hit approval but if there is anything left to address from the comments above then do that first before merging :)

**Comment by sukanyaparashar** [comment]
> cooool - hit approval but if there is anything left to address from the comments above then do that first before merging :)

 Thanks for your approval @eggplantzzz. Yeah! I need to fix few comments and wait for @cliffoo's approval as well.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:34]: Nice 👍

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:90]: It looks like this from [here](https://github.com/trufflesuite/truffle/pull/5320#discussion_r928075653) isn't actually resolved.

Comment (cliffoo) [reply packages/core/lib/commands/init/initSource/truffle-config.js:90]: The whitespace.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:28]: Maybe the require dotenv line should be adjacent to the destructure line, and separated from the hdwallet-provider line? Like so:

```
require dotenv
destructure

hdwallet-provider
```

**Review by cliffoo — APPROVED** [review]
Thanks @sukanyaparashar! This is great. I'm just being picky about a few things, feel free to (not) make changes as you see fit. LGTM!

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:90]: Yeah! I guess I missed that. Will do the required changes. Thanks @cliffoo.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:28]: Yes, we can do this to make `dotenv` usage clearer. I will change this. Thanks for pointing that out @cliffoo.

**Review by cliffoo — APPROVED** [review]
Nice one! Thanks @sukanyaparashar

**Comment by sukanyaparashar** [comment]
> Nice one! Thanks @sukanyaparashar

Thanks for the approval @cliffoo. I am making few more changes to the `truffle-config.js` file comments, after discussing with @cds-amal and @gnidan to make it more appealing to the users to read (like recommending to use `truffle dashboard` for better security practices).

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:35]: The first sentence isn't so clear about what _the_ security and _this_ hassle are. Perhaps this section can be more succinct? Something like:

Are you concerned about security? Truffle dashboard lets you review transactions in detail, and leverages MetaMask for signing so there's no need to copy-paste your mnemonic.

Of course you can insert emojis to your heart's content 🦄

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:90]: Nice catch!

**Review by cliffoo — APPROVED** [review]
Nice one @sukanyaparashar! Definitely a plus to mention dashboard and various trade-offs. My comments are merely suggestions, please proceed with whatever you think is clear and informative 👍

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:27]: I, kind of agree with you @cliffoo. We may not want to directly say `hdwallet-provider` as "insecure". We have already mentioned to add `.env` in `.gitignore`.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (cliffoo) [inline packages/core/lib/commands/init/initSource/truffle-config.js:29]: Final pick, can you remove `the` in this? All good on my side after this ;)

**Review by cliffoo — APPROVED** [review]

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:29]: Oh Okay! Cool! :)

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (gnidan) [inline packages/core/lib/commands/init/initSource/truffle-config.js:29]:
```suggestion
 * Are you concerned about the security 🤔? Truffle Dashboard lets you review transactions in detail, 
```

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (gnidan) [inline packages/core/lib/commands/init/initSource/truffle-config.js:31]: Could you please make sure that all comments fit under 80 characters per line? (My comment here applies generally; this line is just the one that stood out as longest)

**Review by gnidan — DISMISSED** [review]
The new truffle-config.js writeup looks great! Just a couple notes

**Comment by sukanyaparashar** [comment]
> The new truffle-config.js writeup looks great! Just a couple notes

Thanks for the quick review, @gnidan. I will make the suggested changes. :)

**Review by cds-amal — APPROVED** [review]
Thanks for raising security awareness here @sukanyaparashar! Looks good to me.

File: packages/core/lib/commands/init/initSource/truffle-config.js
Comment (gnidan) [inline packages/core/lib/commands/init/initSource/truffle-config.js:14]: Y'know, figuring out whether this should be singular or plural would be a lot easier if this question was correct grammar in the first place... proper grammar here would be "Is your application complex and does it require..." but that sounds so wordy...

Maybe this should be "Do you have a complex application that requires lots of transactions to deploy?" or something that doesn't require combining clauses with an "and".

Comment (gnidan) [reply packages/core/lib/commands/init/initSource/truffle-config.js:14]: Sorry for the poor grammar in the initial wording I wrote for this!

Comment (cds-amal) [reply packages/core/lib/commands/init/initSource/truffle-config.js:14]: > Do you have a complex application that requires lots of transactions to deploy?

I like it

Comment (sukanyaparashar) [reply packages/core/lib/commands/init/initSource/truffle-config.js:14]: No worries! @gnidan. I agree with @cds-amal. I like this "Do you have a complex application that requires lots of transactions to deploy?" wording too. Simple grammar ;)

**Review by gnidan — APPROVED** [review]
have another approve :)

**Review by cliffoo — APPROVED** [review]
Have yet [another one](https://youtu.be/2OVDQ_UVpnE)

**Pull request merged**
